src/main/scala/net/ripe/rpki/publicationserver/store/fs/RepositoryWriter.scala

```diff
@@ -44,63 +35,49 @@ class RepositoryWriter extends Logging {
   def writeSnapshot(rootDir: String, serverState: ServerState, snapshot: Snapshot) = {
     val ServerState(sessionId, serial) = serverState
     val stateDir = getStateDir(rootDir, sessionId.toString, serial)
-    writeFile(snapshot.serialized, new File(stateDir, "snapshot.xml"))
+    writeFile(snapshot.serialized, stateDir.resolve("snapshot.xml"))
   }
 
   def writeDelta(rootDir: String, delta: Delta) = {
     val stateDir = getStateDir(rootDir, delta.sessionId.toString, delta.serial)
-    writeFile(delta.serialize.mkString, new File(stateDir, "delta.xml"))
+    writeFile(delta.serialize.mkString, stateDir.resolve("delta.xml"))
   }
 
   def writeNotification(rootDir: String, notification: Notification) = {
     val root = getRootFolder(rootDir)
 
-    val tmpFile = new File(root, "notification_tmp.xml")
-    writeFile(notification.serialized, tmpFile)
-
-    val source = Paths.get(tmpFile.toURI)
-    val target = Paths.get(new File(root, "notification.xml").toURI)
-    Files.move(source, target, StandardCopyOption.REPLACE_EXISTING)
-  }
-
-  private def writeFile(content: String, file: File) = {
-    val writer = new FileWriter(file)
-    try writer.write(content)
-    finally writer.close()
```

Comment:
Why not?
scala> Files.createTempFile(Paths.get("/tmp"), "notification.", ".xml")
res1: java.nio.file.Path = /tmp/notification.3021118401900893569.xml